contracts/oracle/src/contract/feed.rs

```diff
@@ -77,20 +83,88 @@ where
         Ok(prices)
     }
 
-    fn calc_all_prices(
-        &self,
-        storage: &dyn Storage,
+    pub fn all_prices_iter<'a>(
```

Comment:
this function is more than 80 lines long, contains many sub-scopes and the logic inside is not tested enough

please simplify and *unit* test the code that goes out of the refactoring